src/tuneflow_py/models/lyric.py

```diff
@@ -0,0 +1,421 @@
+            self._proto.start_tick = start_tick
+            self._proto.end_tick = end_tick
+
+    def get_line(self):
+        return self.line
+
+    def get_word(self):
+        return self._proto.word
+
+    def set_word(self, word: str):
+        if not word:
+            raise ValueError("Word cannot be empty")
+        self._proto.word = word
+
+    def get_start_tick(self):
+        return self._proto.start_tick
+
+    def set_start_tick(self, start_tick: int, resolve_order=True):
+        '''
+        Set a new start tick for the word.
+        This might delete the word if the new start tick is greater than or equal to the end tick.
+
+        Args:
+            start_tick (int): The new start tick of the word.
+        '''
+        if self._proto.start_tick >= self._proto.end_tick and resolve_order:
+            self.delete_from_parent()
+            return
+        self._proto.start_tick = start_tick
+        if resolve_order:
+            self.line.sort_words()
+
+    def get_end_tick(self):
+        return self._proto.end_tick
+
+    def set_end_tick(self, end_tick: int, resolve_order=True):
+        '''
+        Set a new end tick for the word.
+        This might delete the word if the new end tick is less than or equal to the start tick.
+
+        Args:
+            end_tick (int): The new end tick of the word.
+        '''
+        if end_tick <= self._proto.start_tick and resolve_order:
+            self.delete_from_parent()
+            return
+        self._proto.end_tick = end_tick
+
+    def move_to(self, start_tick: int, end_tick: int):
+        self._proto.start_tick = start_tick
+        self._proto.end_tick = end_tick
+        self.line.sort_words()
+
+    def delete_from_parent(self):
+        '''
+        Delete the word from the parent line.
+        '''
+        self.line.remove_word(self)
+
+    def __eq__(self, other: LyricWord):
+        if isinstance(other, song_pb2.LyricLine.LyricWord):
+            return (
+                self._proto.start_tick == other.start_tick
+                and self._proto.end_tick == other.end_tick
+                and self._proto.word == other.word
+            )
+        elif isinstance(other, LyricWord):
+            return (
+                self._proto.start_tick == other._proto.start_tick
+                and self._proto.end_tick == other._proto.end_tick
+                and self._proto.word == other._proto.word
+            )
+        return False
+
+
+class LyricLine:
+    '''
+    Lyrics is composed of a list of LyricLine objects
+    The class is a wrapper of the LyricLine proto object
+    '''
+
+    def __init__(
+        self,
+        lyrics: Lyrics,
+        start_tick: int | None = None,
+        proto: song_pb2.LyricLine | None = None
+    ):
+        self.lyrics = lyrics
+        if proto is not None:
+            self._proto = proto
+        else:
+            self._proto = song_pb2.LyricLine()
+
+        if len(self._proto.words) == 0:
```

Comment:
I guess this is only needed when proto is not provided.

Reply:
it's used when we add() a new line proto without repeating creating the placeholder within create_line or create_line_from_string. so it ensures there's always a placeholder